### PR TITLE
Don’t run `bypass-checks` on non "run" actions

### PR DESCRIPTION
**Pull request by yakov116**
1. LINKED ISSUES:
   None

2. TEST URLS:
	https://github.com/eslint/eslint/pull/13748
3. SCREENSHOT:
   None

### CODE REVIEW

**Comment by yakov116** [comment]
I dont like what I wrote. maybe will gibve it a shot in the morning. @fregante I am open to idea's.

**Comment by fregante** [comment]
> 2\. TEST URLS:
>     [eslint/eslint#13748](https://github.com/eslint/eslint/pull/13748)

Which one? I don't see any check that needs to be bypassed there.

File: source/features/bypass-checks.tsx [hunk 1]
```diff
@@ -7,7 +7,13 @@ import * as api from '../github-helpers/api';
 import {getRepoURL} from '../github-helpers';
 
 async function bypass(detailsLink: HTMLAnchorElement): Promise<void> {
-	const runId = new URLSearchParams(detailsLink.search).get('check_run_id') ?? detailsLink.pathname.split('/').pop()!;
+	const [actionType, splitRunid] = detailsLink.pathname.split('/').splice(-2);
+	const runId = new URLSearchParams(detailsLink.search).get('check_run_id') ?? splitRunid;
+
+	if (!Number.parseInt(runId, 10) || actionType !== 'run') {
```
Comment (fregante) [inline source/features/bypass-checks.tsx:13]: A bit verbose. Can pageDetect help?

Comment (yakov116) [reply source/features/bypass-checks.tsx:13]: I am 100% at fault here. I wrote this half in my sleep, I should have waited for the morning...

I will update it later with more details

**Comment by yakov116** [comment]
Let me explain the issue.
https://github.com/eslint/eslint/pull/13748 has a check for https://github.com/eslint/eslint-github-bot/blob/master/docs/commit-message-check.md and https://github.com/eslint/eslint/issues/13719 (hover over the first commit status to see the second one)

These 2 are not action runs and they already link to their direct page.

**Comment by fregante** [comment]
Use GitHub URL Detection with an argument, no need for custom reflexes unless you’re *extracting* data. In this case `isActionRun(detailsLink)` should work, except that the types don’t accept a raw HTMLAnchorElement. PR welcome there to change that

**Comment by fregante** [comment]
Also include the 2 types of URLs in comments

File: source/features/bypass-checks.tsx [hunk 2]
```diff
@@ -7,7 +7,12 @@ import * as api from '../github-helpers/api';
 import {getRepoURL} from '../github-helpers';
 
 async function bypass(detailsLink: HTMLAnchorElement): Promise<void> {
-	const runId = new URLSearchParams(detailsLink.search).get('check_run_id') ?? detailsLink.pathname.split('/').pop()!;
+	const runId = pageDetect.isActionJobRun(detailsLink) ? 	detailsLink.pathname.split('/').pop() :// https://github.com/xojs/xo/runs/1104625522
```
Comment (fregante) [inline source/features/bypass-checks.tsx:10]: Rough whitespace around here. There's a tab after `?`

This is the expected indentation for the ternary operator
```suggestion
	const runId = pageDetect.isActionJobRun(detailsLink) ?
		detailsLink.pathname.split('/').pop() : // https://github.com/xojs/xo/runs/1104625522
```

File: source/features/bypass-checks.tsx [hunk 3]
```diff
@@ -7,7 +7,12 @@ import * as api from '../github-helpers/api';
...
+		new URLSearchParams(detailsLink.search).get('check_run_id'); // https://github.com/sindresorhus/refined-github/pull/3629/checks?check_run_id=1223857819
+	if (!runId) {
+		return;
```
Comment (fregante) [inline source/features/bypass-checks.tsx:13]:
```suggestion
		// Sometimes the URL doesn't point to Checks at all
		return;
```

**Review by fregante — APPROVED** [review]

**Comment by fregante** [comment]
You can drop the package files, I updated the dependency in #3639

**Pull request merged**